### PR TITLE
Expense Flow: Payee information as First Step

### PR DESCRIPTION
**Pull request by kewitz**
Resolves https://github.com/opencollective/opencollective/issues/3495

**Out of scope update:**
I had to move the minimum transfer amount for selected currency error to the end of the form, previously this was displayed in the currency dropdown but it no longer makes sense since this is now on page 1.
image

### CODE REVIEW

File: components/expenses/ExpenseForm.js [hunk 1]
```diff
@@ -76,26 +78,30 @@ const msg = defineMessages({
     id: 'ExpenseForm.AddLineItem',
     defaultMessage: 'Add new item',
   },
-  step1: {
-    id: 'ExpenseForm.Step1',
+  stepReceipt: {
```
Comment (Betree) [inline components/expenses/ExpenseForm.js:81]: Way better name :+1:

File: components/expenses/ExpenseForm.js [hunk 2]
```diff
@@ -215,6 +221,15 @@ const refreshPayoutProfile = (formik, payoutProfiles) => {
   formik.setFieldValue('payee', payee);
 };
 
+const HiddenFragment = styled.div`
+  display: ${({ show }) => (show ? 'block' : 'none')};
```
Comment (Betree) [inline components/expenses/ExpenseForm.js:225]: Good call to do that, hiding with CSS is way more performant. Having the form displayed in JS when you typed the first character could freeze a bit the interface.

**Review by Betree — COMMENTED** [review]
The code looks really clean, great job!

The only problem that I found is this:
- Pick `Reimbursement` as the type
- Use a profile that doesn't have an address yet (because it's a receipt, it won't ask you to enter one)
- Click next
- Fill some stuff
- Scroll top, try to change the type to Invoice
- Form is stucked because there's no payee address

Maybe changing the type should take you back to step 1 (but ideally we want to keep the data entered)

File: components/expenses/ExpenseForm.js (hunk 2 above)
Comment (kewitz) [reply components/expenses/ExpenseForm.js:225]: Yes, I was also having trouble with Formik since fields are not rendered as you can imagine.

**Pull request merged**